packages/cells/src/model.ts

```diff
@@ -620,3 +651,33 @@ export namespace CodeCellModel {
    */
   export const defaultContentFactory = new ContentFactory();
 }
+
+namespace Private {
+  export function collapseChanged(
+    metadata: IObservableJSON,
+    args: IObservableMap.IChangedArgs<JSONValue>
+  ) {
+    if (args.key === 'collapsed') {
+      const jupyter = (metadata.get('jupyter') || {}) as JSONObject;
+      const { outputs_hidden, ...newJupyter } = jupyter;
+
+      if (outputs_hidden !== args.newValue) {
+        if (args.newValue !== undefined) {
+          newJupyter['outputs_hidden'] = args.newValue;
+        }
+        if (Object.keys(newJupyter).length === 0) {
+          metadata.delete('jupyter');
+        } else {
+          metadata.set('jupyter', newJupyter);
+        }
+      }
+    } else if (args.key === 'jupyter') {
+      const jupyter = (args.newValue || {}) as JSONObject;
+      if (jupyter.hasOwnProperty('outputs_hidden')) {
+        metadata.set('collapsed', jupyter.outputs_hidden);
+      } else {
+        metadata.delete('collapsed');
+      }
+    }
+  }
+}
```

Comment:
I think there may be something wrong in this logic. Here is the behavior I am seeing:

1. Set a cell's metadata to `"collapsed": true` and see `outputs_hidden` get created.
1. Set `"collapsed": false` and save and it comes back again as `true`
1. Set `"collapsed": false` again and metadata block disappears.

![metadata](https://user-images.githubusercontent.com/159529/55577477-93f3dc80-570b-11e9-8d1d-5fec4c6f1f75.gif)

Reply:
When I am in the cell metadata editor, if I hit `Accel S` the notebook file saves, but the actually metadata I am trying to save doesn't save unless I click on the checkbox button. Do you think that should be fixed in a separate PR? I'll open an issue if so.

Reply:
Thanks - I agree it's a separate issue.

Reply:
Cool.

Is the behavior in the screen cap happening for you as well?

Reply:
> Is the behavior in the screen cap happening for you as well?

I haven't checked yet.

Reply:
Yes, I can reproduce the issue you saw. Investigating...

Reply:
Turns out that the metadata editor was setting *all* fields, instead of just setting fields that changed. That meant that when it set `collapsed`, the `outputs_hidden` was being updated properly, but then the metadata editor would set `outputs_hidden` to its old value, which would cause collapsed to be changed. I fixed the metadata editor to only set values that had actually changed from the original value.